tests/rptest/tests/data_transforms_test.py

```diff
@@ -672,4 +684,212 @@ def test_tunable_configs(self):
+                   timeout_sec=30,
+                   backoff_sec=5,
+                   err_msg=f"never got all the events")
+
+        totals, dropped = get_total_events_per_xform()
+        assert len(totals) == len(its)
+        for name in totals:
+            assert totals[
+                name] >= N_PER_TP, f"Expected {N_PER_TP} log events from transform {name}, got{totals[name]}"
+        assert dropped == 0, f"Expected no dropped events, got {dropped}"
+
+    @cluster(num_nodes=3)
+    def test_manager_metrics_present(self):
+        it, _ = self.setup_identity_xform(self.topics[0], self.topics[1])
+
+        def has_manager_metrics(node, endpoint):
+            metrics_samples = self.get_metrics_from_node(node,
+                                                         self.MANAGER_METRICS,
+                                                         endpoint=endpoint)
+            assert metrics_samples is not None, "Get_metrics timed out"
+            return sorted(metrics_samples.keys()) == sorted(
+                self.MANAGER_METRICS)
+
+        assert all(
+            [
+                has_manager_metrics(n, MetricsEndpoint.METRICS)
+                for n in self.redpanda.nodes
+            ]
+        ), "Expected all nodes to export manager metrics on the internal endpoint"
+
+        assert not any([
+            has_manager_metrics(n, MetricsEndpoint.PUBLIC_METRICS)
+            for n in self.redpanda.nodes
+        ]), "Expected no node to export manager metrics on the public endpoint"
+
+    @cluster(num_nodes=3)
+    def test_manager_metrics_values(self):
+        self.logger.debug("Set flush interval to 1h so buffers fill up")
+        self.redpanda.set_cluster_config(
+            {'data_transforms_logging_flush_interval_ms': 60 * 60 * 1000})
+
+        it, ot = self.setup_identity_xform(
+            self.topics[0],
+            self.topics[1],
+            name=f"logger-xform",
+        )
+
+        self.logger.debug(
+            "Produce enough data to make a noticeable dent but not so much as to trip the buffer LWM"
+        )
+        N_PER_TP = 256
+        for _ in range(0, N_PER_TP):
+            self._rpk.produce(
+                it.name,
+                self.random_ascii_string(100, 100),
+                self.random_ascii_string(100, 100),
+            )
+
+        self._rpk.consume(ot.name, n=1, offset=N_PER_TP - 1)
+
+        all_nodes_usage = []
+        for node in self.redpanda.nodes:
+            samples = self.unpack_samples(
+                self.get_metrics_from_node(
+                    node,
+                    ["log_manager_buffer_usage"],
+                    endpoint=MetricsEndpoint.METRICS,
+                ))
+
+            for s in samples['log_manager_buffer_usage']:
+                all_nodes_usage.append(s['value'])
+
+        assert any(
+            bu > 0.0 for bu in all_nodes_usage
+        ), f"Expected some non-zero buffer usage, got {all_nodes_usage}"
```

Comment:
Is this racy in that all the logs could be flushed before we query the metrics?

Reply:
I don't think so. It is timing dependent, but the flush interval is configured to 1h at the top of the test. Or are you concerned that we're racing against the config binding watcher?